test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageExtraction/StreamExtensionsTests.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Text;
+using Moq;
+using NuGet.Common;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Packaging.Test.PackageExtraction
+{
+    class TestableProxy : NuGet.Packaging.StreamExtensions.Testable
```

Comment:
private classes should be nested inside the class that uses them

```cs
public class WhateverTests
{
  private class TestableProxy : Thing
  {
  }
}
```

so as to not pollute Intellisense with this class in all other files. Or if you think the class is useful for other classes, then the name needs to be better and put in a separate file (one class per file), because anyone writing other tests in this project won't have a clue `TestableProxy` is for when outside the context of this file. But I personally don't think that this class will be useful to other classes in the project, so I suggest just make it nested.